doc/qrexec-remotevm.rst

```diff
@@ -0,0 +1,173 @@
+RemoteVM with the name ``Remote-Qube``, resulting in the following sequence diagram:
+
+.. image:: _static/particular.png
+
+SSH-based communication between Local‑Relay and Remote‑Relay
+------------------------------------------------------------
+
+Prerequisites
+~~~~~~~~~~~~~
+
+- SSH key-based authentication is configured between all involved hosts (Local‑Relay, Remote‑Relay, and any intermediate jump hosts).
+- (Optional) Configure any necessary proxies or jump hosts by following the OpenSSH Cookbook guidelines: Proxies and Jump Hosts <https://en.wikibooks.org/wiki/OpenSSH/Cookbook/Proxies_and_Jump_Hosts>.
+
+Example Transport RPC: ``qubesair.SSHProxy``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Below is an example Bash script implementing the ``qubesair.SSHProxy`` transport RPC. It parses a destination and service, looks up the remote Qube name, and invokes ``qrexec-client-vm`` over SSH.
+
+.. code-block:: bash
+
+   #!/bin/bash
+
+   set -euo pipefail
+
+   if [[ -z "${1:-}" ]]; then
+       echo "Usage: $0 <target+service>"
+       exit 1
+   fi
+
+   # Extract target and service
+   IFS='+' read -r target service <<< "$1"
+
+   # Validate extracted values
+   if [[ -z "$target" || -z "$service" ]]; then
+       echo "Error: Invalid input format. Expected 'target+service'."
+       exit 1
+   fi
+
+   # Retrieve the remote Qube identifier
+   remote_qube="$(qubesdb-read "/remote/$target" 2>/dev/null || true)"
+
+   if [[ -z "$remote_qube" ]]; then
+       echo "Error: Could not read remote qube for target '$target'."
+       exit 1
+   fi
+
+   # Forward the qrexec call via SSH
+   ssh "$remote_qube" qrexec-client-vm --source-qube="$QREXEC_REMOTE_DOMAIN" "$remote_qube" "$service"
+
+Workflow Overview
+"""""""""""""""""
+
+1. **Local-Qube to Local-Relay**: A Qubes RPC call is made from Local-Qube to Local-Relay with arguments of the form:
+
+   ``Remote-Qube+<service>``
+
+    where `<service>` contains the original service with argument that has been processed by policies.
+
+2. **Destination lookup**: The RPC script extracts `Remote-Qube` and queries QubesDB on Local‑Relay to verify and translate it into the actual qube name in `Remote-QubesOS` (in this example it is assumed to be the same).
+
+3. **SSH**: The script uses SSH to connect to the identified host (configured to forward to the Remote‑Relay). It then runs ``qrexec-client-vm`` on that host, passing along the original source domain via the environment variable ``$QREXEC_REMOTE_DOMAIN``.
+
+4. **Remote-QubesOS policy evaluation**: `Remote‑QubesOS` applies its policy rules for a request from `Local‑Relay` to `Remote‑Qube` with information that source qube is `Local-Qube`.
+
+SSH Client Configuration Hint
++++++++++++++++++++++++++++++
+
+To ensure that any SSH connection to a given host actually lands on the corresponding `Remote‑Relay`, add an entry like the following to your ``/etc/ssh/ssh_config``:
```

Comment:
Maybe better `~/.ssh/config`? It will persist without extra bind-dirs then.

As for the hints - on the remote site /etc/ssh better be made persistent with bind-dirs (and the host key re-generated there), otherwise host key will be shared across all VMs based on the same template, so not a secret anymore...

Reply:
I know but for me it's up to the user to know that's configuring host key at system level (ofc we would document it in qubes-doc at the end). Also for home path, is RPC using user home?

Reply:
Services are called as the default user normally, so yes, it will be `user`.

Reply:
Yes let's ignore with a warning.